File: src/main/webapp/resources/core/js/services/ClienteService.js
function ClienteService(){
	
	var _add = function(cliente){
		return $.get(SERVER_APP + '/cliente/add', cliente);
	}
	
	var _findByName = function(nome){
		return $.get(SERVER_APP + '/cliente/findByName/'+ nome);
	}
		
	return {

		add: _add,
		
		findByName: _findByName
		
	}
}

File: src/main/java/br/com/estoquemary/services/ClienteService.java
package br.com.estoquemary.services;

import java.util.List;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import br.com.estoquemary.models.Cliente;
import br.com.estoquemary.repositories.ClienteRepository;

@Service
public class ClienteService {

	@Autowired
	private ClienteRepository clienteRepository;
	
	public Cliente save(Cliente cliente) {
		return clienteRepository.save(cliente);
	}

	public List<Cliente> findByName(String nome) {
		return clienteRepository.findByNomeContainingIgnoreCase(nome);
	}

	public List<Cliente> findAll() {
		return (List<Cliente>) this.clienteRepository.findAll();
	}

}
